### PR TITLE
Allow users to add the bot to any chat

### PR DESCRIPTION
**Pull request by martijnjanssen**

### CODE REVIEW

File: F1ScheduleTelegramBot/main.py [hunk 1]
```diff
@@ -29,23 +30,68 @@
...
+    else:
+        name = update.effective_chat.title
+
+    # disallow using the dev name to start the chat
+    if name == DEV_CHAT_NAME:
+        await context.bot.send_message(
+            chat_id=chat_id,
+            text="Hello! I am F1ScheduleTelegramBot! I am currently unable to work in "
+            "this chat, since the chat name `{}` is reserved".format(DEV_CHAT_NAME),
+        )
```
Comment (Fastjur) [inline F1ScheduleTelegramBot/main.py:53]: Euhm, why? I'd rather not do this or do this based on a chat_id if we really want to add this. What is the reason for this?

File: F1ScheduleTelegramBot/main.py [hunk 2]
```diff
@@ -29,23 +30,68 @@
...
+
+    try:
+        database.get_chat(dbconn, chat_id)
+        await context.bot.send_message(
+            chat_id=chat_id,
+            text="Your chat has already been registered! {}{}{}".format(EMOJI_RED_FLAG,
+                                                                        EMOJI_RED_FLAG,
+                                                                        EMOJI_RED_FLAG),
```
Comment (Fastjur) [inline F1ScheduleTelegramBot/main.py:61]: Why not just put it in the string here?

File: F1ScheduleTelegramBot/main.py [hunk 3]
```diff
@@ -29,23 +30,68 @@
...
+        return
+    except Exception:
+        cur = dbconn.cursor()
+        cur.execute("INSERT INTO chats VALUES (:chat_id, :type, :name)",
+                    {
+                        "chat_id": chat_id,
+                        "type": update.effective_chat.type,
+                        "name": name,
+                    })
+        dbconn.commit()
```
Comment (Fastjur) [inline F1ScheduleTelegramBot/main.py:72]: Too broad exception catch to be handling insertions into db IMO.

File: F1ScheduleTelegramBot/main.py [hunk 4]
```diff
@@ -29,23 +30,68 @@
...
+
     await context.bot.send_message(
-        chat_id=update.effective_chat.id,
-        text="Hello! I am F1ScheduleTelegramBot! I am currently mostly hardcoded, but more "
-             "features will be coming soon! Your chat id is: `{}`".format(update.effective_chat.id),
+        chat_id=chat_id,
+        text="Hello! I am F1ScheduleTelegramBot. I am currently mostly hardcoded, but more "
+             "features will be coming soon!\n\n"
+             "Your chat has been registered successfully {}".format(EMOJI_CHEQUERED_FLAG),
```
Comment (Fastjur) [inline F1ScheduleTelegramBot/main.py:78]: Why not just add the emoji in the string?

File: F1ScheduleTelegramBot/main.py [hunk 5]
```diff
@@ -130,33 +170,28 @@ def main():
...
-    chats = database.list_chats(dbconn)
-    if len(chats) == 0:
-        cur.execute("INSERT INTO chats VALUES (:name, :chat_id)",
-                    {"name": "CHAT", "chat_id": chat_id})
+        cur.execute("INSERT INTO chats VALUES (:chat_id, :type, :name)",
+                    {
+                        "chat_id": chat_id_dev,
+                        "type": telegram.constants.ChatType.GROUP,
+                        "name": DEV_CHAT_NAME,
+                    })
```
Comment (Fastjur) [inline F1ScheduleTelegramBot/main.py:194]: How safe is this? There's no user input here at all right?

Comment (martijnjanssen) [reply F1ScheduleTelegramBot/main.py:194]: There is no user input, just keeping things uniform

File: F1ScheduleTelegramBot/main.py (hunk 1 above)
Comment (martijnjanssen) [reply F1ScheduleTelegramBot/main.py:53]: I'll remove it, was a case of us being able to see what the dev id was.

File: F1ScheduleTelegramBot/main.py (hunk 3 above)
Comment (martijnjanssen) [reply F1ScheduleTelegramBot/main.py:72]: Filed #30

**Review by Fastjur — APPROVED** [review]

**Pull request merged**